packages/gateway/src/gateway-embedding-model.ts

```diff
@@ -91,4 +91,5 @@ export class GatewayEmbeddingModel implements EmbeddingModelV2<string> {
 const gatewayEmbeddingResponseSchema = z.object({
   embeddings: z.array(z.array(z.number())),
   usage: z.object({ tokens: z.number() }).nullish(),
+  providerMetadata: z.any().optional(),
```

Comment:
Avoid use of `any` -- revise to type this, look at how we handle it for `generateText`, `generateObject`, etc. It is probably set up as a `Record<string, unknown>` or similar.